### ISSUE TITLE
Merge update.

### ISSUE THREAD

**Issue by jdalton** [underdash/underdash#14]
Hey Ya'll,  (TL;DR is at the bottom of this comment)

I wanted to give an update on the merge discussions.

On January 12 we released [lodash v4](https://github.com/lodash/lodash/releases/tag/4.0.0). From its release notes:

>  The year was also one of collaboration, as discussions began on merging Lodash & Underscore. Much of Lodash v4 is proofing out the ideas from those discussions. Lodash v4 **would not be possible** without the collaboration & contributions of the Underscore core team. In the spirit of merging our teams have blended with [several members](https://github.com/orgs/lodash/people) contributing to both libraries.

It bears repeating Lodash v4 **would not be possible** without the collaboration & contributions of the Underscore core team. Since the merge talks began 8 months ago, with our now blended core, and combined efforts, Lodash has continued to gain ground. The following stats were reviewed by core members of both libs and are meant to show how far we've come as a whole (blended core), where our cumulative priorities are, and where the community is. (Sorry for the stats dump)

From non-core contributions (taken 2015-Feb. 2016):

<table>
<tr>
<th>Lodash</th><th>Underscore</th>
<tr>
<td>
~131 contributions
</td>
<td>
~96 contributions
</td>
</table>

to number of releases (taken 2015-Feb. 2016):

<table>
<tr>
<th>Lodash</th><th>Underscore</th>
<tr>
<td>
23 releases
</td>
<td>
4 releases
</td>
</table>

to number of commits (taken 2015-Feb. 2016):

<table>
<tr>
<th>Lodash</th><th>Underscore</th>
<tr>
<td>
1,977 commits
</td>
<td>
422 commits
</td>
</table>

to repo traffic (taken Feb. 2016):

<table>
<tr>
<th>Lodash</th><th>Underscore</th>
<tr>
<td colspan=2>Lodash has 2.8x clones, 1.8x uniq clones, 5.6x views, & 3x the uniq visitors of Underscore</td>
</tr>
<tr>
<td>
<img src="https://cloud.githubusercontent.com/assets/4303/13024167/18ffec00-d1a6-11e5-9070-e0f2c21e15ec.png" title="Click to enlarge">
</td>
<td>
<img src="https://cloud.githubusercontent.com/assets/4303/13024180/4292dd7a-d1a6-11e5-9806-282f794984e2.png" title="Click to enlarge">
</td>
</table>

to surveys:

<table>
<tr>
<td>
backbone survey
</td>
</tr>
<tr>
<td>
<a href="https://twitter.com/jdalton/status/696042215617900544"><img src=https://cloud.githubusercontent.com/assets/4303/13024118/809b8672-d1a5-11e5-8208-8bbdd40d5299.png width=80%>
</td>
</tr>
<tr>
<td>
developer survey
</td>
</tr>
<tr>
<td>
<a href="https://twitter.com/jdalton/status/675157693284392960"><img src="https://cloud.githubusercontent.com/assets/4303/13024130/a0d3c9ea-d1a5-11e5-9924-9f881700ce60.png" height=60%></a>
</td>
</tr>
</table>

to npm package dependents (taken Feb. 2016):

<table>
<tr>
<td>
<a href="https://gist.github.com/anvaka/8e8fa57c7ee1350e3491#file-01-most-dependent-upon-md
"><img src="https://cloud.githubusercontent.com/assets/4303/13024267/1d06dd30-d1a7-11e5-8276-94f3ce29bd7a.png" width=80%></a>
</td>
</tr>
</table>

dependent growth compared to npm's registry growth (taken Jul. 2015):

<table>
<tr>
<td>
<a href="https://twitter.com/seldo/status/625874919876861952"><img src=https://cloud.githubusercontent.com/assets/4303/13024296/821b3d24-d1a7-11e5-9081-6557227f85b4.png width=85%></a>
</td>
</tr>
</table>

and growth per month (taken Oct. 2015):

<table>
<tr>
<td>
Lodash is the fastest growing npm package with ~1,000 new dependents each month
</td>
</tr>
<tr>
<td>
<a href="https://anvaka.github.io/npmrank.vis/">
<img src="https://cloud.githubusercontent.com/assets/4303/13024410/3ed8c610-d1a9-11e5-96e3-bb20174ef832.png" width=40%></a>
</td>
</tr>
</table>

even total commits year over year (taken Feb. 2016):

<table>
<tr>
<th>Lodash (up 10%)</th><th>Underscore (down 65%)</th>
<tr>
<td>
<a href="https://msit.powerbi.com/"><img src="https://cloud.githubusercontent.com/assets/4303/13025394/37512380-d1bb-11e5-9bc8-3e6b877219a4.png" width=50%></a>
</td>
<td>
<a href="https://msit.powerbi.com/"><img src="https://cloud.githubusercontent.com/assets/4303/13025393/3750bc9c-d1bb-11e5-874d-fd56ba39cf77.png" width="50%"></a>
</td>
</table>

With Jeremy [taking a step back](https://github.com/jashkenas), Underscore core like @braddunbar @jridgewell @megawac already contributing to Lodash, combined with the community and momentum it makes sense to continue to focus our energy on Lodash.
#### TL;DR

So as to the merge _I think_ it's already happened as Lodash v4. We share core contributors, @jridgewell has npm publish rights for Backbone/Lodash/Underscore, and the community has moved along with us. Thanks everyone for your input, comments, suggestions, and support!

**Comment by timdorr** [comment]
This is great. Thanks for all the detail. 

Is there anywhere where help is needed? Or are there already enough cooks in the kitchen?

**Comment by jdalton** [comment]
@timdorr 
There's plenty to help with on the Lodash side. We need help fleshing out our [FP guild](https://github.com/lodash/lodash/wiki/FP-Guide) as well as help with [site design/maintenance](https://github.com/lodash/lodash.github.io/issues/8). Several of [our projects](https://github.com/lodash) have `help wanted` issue labels to identify areas to pick up.

**Comment by vendethiel** [comment]
@jdalton you wrote "fp **guild**" ;)

**Comment by akre54** [comment]
Anyone surprised? Bueller?

**Comment by thejameskyle** [comment]
@akre54 Thanks for the meaningful contribution to this discussion.

**Comment by akre54** [comment]
![](http://gifrific.com/wp-content/uploads/2014/02/Michelle-Tanner-You-Got-it-Dude-Full-House.gif)

**Comment by jmeas** [comment]
Any planz to mark Underscore as deprecated?

**Comment by akre54** [comment]
Nope, not deprecated. Still pretty much status quo ante as far as I'm concerned.

**Comment by jdalton** [comment]
@jmeas 
> Any planz to mark Underscore as deprecated?

That hasn't been discussed in depth yet. One of the benefits of these merge discussions was it avoided making the call of ending one or the other. While not officially deprecated I imagine Underscore's activity/usage will continue its slow decline. However, deprecation and migration is an Underscore core call and not something Lodash core will jump into.

**Comment by akre54** [comment]
I could go through point-by-point about how ridiculous your statistics and surveys are, but none of that really matters to you anyway (or anyone else, apparently). Just keep on keepin on.

**Comment by jdalton** [comment]
@akre54
>  but none of that really matters to you anyway (or anyone else, apparently)

Yeah, it's your hang-up man. It'd be great if you turned that frustration into something productive. You've had only [6 non-merge commits](https://github.com/jashkenas/underscore/graphs/contributors?from=2015-06-1&to=2016-02-17&type=c)  to Underscore in the last 8 months. There's plenty of [open issues](https://github.com/jashkenas/underscore/issues?utf8=%E2%9C%93&q=+is%3Aopen+) and discussions like `underdash` you could have participated in constructively. Grumbling after the fact is a bit too little too late.

> Just keep on keepin on.

Everyone else has. I'd be great if you pitched in too.

**Comment by michaelficarra** [comment]
Please stop, both of you. Comparing the number of commits you've made or how many downloads/stars/high-fives your library has is childish. This is OSS. If you like working on something, work on it. Save the banter for Twitter.

**Comment by jdalton** [comment]
> Please stop, both of you. Comparing the number of commits you've made or how many downloads/stars/high-fives your library has is childish. 

I don't think it was quite like that. Establishing the position of libs and the changing landscape is necessary. There's been a lot of collaboration between core with folks like @braddunbar, @jridgewell, & @megawac. Now you and @akre54 weren't really a part of it but that's cool 3 out of 5 ain't bad.

**Comment by jdalton** [comment]
You aren't continuing Underscore, you're not putting in the time to keep it going, to keep it healthy, neither is Adam. You both aren't willing to do what's needed to keep the project going. But you jump in and try to rain on progress of the others.

**Comment by jdalton** [comment]
You're input isn't really needed as its been absent the entire time.
